--- unix.cmake	2025-01-22 12:59:52.050127544 +0700
+++ ./cmake/platforms/unix.cmake	2025-01-22 15:12:22.388879851 +0700
@@ -109,15 +109,15 @@
       # CMAKE_INSTALL_BINDIR.
       install(TARGETS ${TARGET})
     endif()
-    configure_file(${CMAKE_SOURCE_DIR}/puzzle.desktop.in ${binary_name}.desktop)
+    configure_file(${CMAKE_SOURCE_DIR}/puzzle.desktop.in uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}.desktop)
     foreach(icon_size ${all_icon_sizes})
       install(
         FILES ${CMAKE_CURRENT_BINARY_DIR}/icons/${NAME}-${icon_size}d24.png
         DESTINATION share/icons/hicolor/${icon_size}x${icon_size}/apps
         OPTIONAL
-        RENAME ${binary_name}.png)
+        RENAME uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}.png)
     endforeach()
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${binary_name}.desktop
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}.desktop
       DESTINATION share/applications)
   endif()
 endfunction()
--- puzzle.desktop.in	2025-01-22 12:59:52.067127580 +0700
+++ ./puzzle.desktop.in	2025-01-22 15:21:24.837795300 +0700
@@ -4,7 +4,7 @@
 Name=${displayname}
 Comment=${description}
 Exec=${binary_name}
-Icon=${binary_name}
+Icon=uk.org.greenend.chiark.sgtatham.puzzles.${binary_name}
 StartupNotify=false
 Categories=Game;LogicGame;
 Terminal=false
